test/e2e/es_cluster_logging.go

```diff
@@ -139,14 +139,20 @@ func ClusterLevelLoggingWithElasticsearch(f *Framework) {
 	// Now assume we really are talking to an Elasticsearch instance.
 	// Check the cluster health.
 	By("Checking health of Elasticsearch service.")
-	body, err := f.Client.Get().
-		Namespace(api.NamespaceDefault).
-		Prefix("proxy").
-		Resource("services").
-		Name("elasticsearch-logging").
-		Suffix("_cluster/health").
-		Param("health", "pretty").
-		DoRaw()
+	var body []byte
+	for start := time.Now(); time.Since(start) < graceTime; time.Sleep(5 * time.Second) {
```

Comment:
At some point a single grace period was forced upon all the loops in this test -- something I did not like. This is because some retry loops should take a short amount of total time (all the ones up to the last one) and some loops need longer e.g. to gather the logs (the last one). Before the retry timeouts were adjusted based on the functionality being tested.